### DISCUSSION TITLE
Arch.Extended ParallelQuery source generator?

### DISCUSSION THREAD

**Discussion by stgeorge** [genaray/Arch#62] [Ideas]
This is more for the Arch.Extended project but it didn't have a Discussions section.

I noticed there's not an easy way to change a [Query] to parallel other than kind of pulling it out of the source generator and using the Arch scheme. While this is OK, it would be pretty cool to just be able to say [ParallelQuery] and have it work without much fuss. I am liking the source generator approach in terms of less boilerplate.

**Comment by genaray** [comment]
Oh sorry, havent seen this :D 

This is actually planned, didn't had time to implement this yet. But it would actually be quite easy, find a method annotated with such a attribute, copy its content into a struct capable of being multithreaded and then the generated method would just need to push it into the parallel queries API :)
